Automattic/jetpack
# Social: Add Auto-Conversion feature UI toggles

This change adds the UI toggles to the Social admin page and under Jetpack Settings, for our upcoming auto-conversion feature.

<img width="886" alt="CleanShot 2023-08-21 at 11 38 34 png" src="https://github.com/Automattic/jetpack/assets/36671565/e5dbac5d-961a-4128-9871-a0b750eec73d">

<img width="1066" alt="CleanShot 2023-08-21 at 11 38 47 png" src="https://github.com/Automattic/jetpack/assets/36671565/b59ab1d6-3ce2-4dff-8148-5792f6495132">

## Proposed changes:

* Added the needed initial state changes in b055a20ec69a7df13d0bc744ae6cfe602fbd1696
* Extended the Jetpack Social store in `publicize-components` to be able to persist the changes for the auto-conversion feature 1ee71f353ea856161905aa043f1b3561b76d2196
* Added the actual toggle to the Social Admin page 55534ce24785c571f6522c9f5ac1bfda2fd376d7
* Added the toggle under Jetpack Settings cf7eafe86e98006bf55e228ecb007512b7a253bc

### Other information:


## Jetpack product discussion


https://github.com/orgs/Automattic/projects/733/views/2?pane=issue&itemId=36021083
## Does this pull request change what data or activity we track or use?


No.
## Testing instructions:






* Open up the Jetpack Social admin page, and the Jetpack Settings page's sharing section
  * You should not see the auto-conversion toggle anywhere
* Add the `jetpack-social-image-auto-convert` blog sticker to your site
  * Now you should be able to see the toggles.
* Test the functionality of the toggles
  * Toggling the state on/off should be reflected on the UI, and should stay persisted after browser refresh
  * Change the state in the Social admin page, then refresh the page on the Jetpack Settings, it should get updated too. Do it the other way around.
* Check the help URL-s it should redirect you to the support page for the media restrictions.
